api/envoy/extensions/filters/http/credential_injector/v3/credential_injector.proto

```diff
@@ -74,12 +74,14 @@ message CredentialInjector {
   // Value defaults to false.
   bool overwrite = 1;
 
-  // Whether to fail the request if the credential is not present.
+  // Whether to fail the request if the credential is not present or failed to inject the credential
+  // to the request.
   // Value defaults to false.
-  // If set to true, the request will fail with 401 Unauthorized if the credential is not present.
-  bool fail_if_not_present = 2;
+  // If set to true, the request will fail with 401 Unauthorized if the credential is not present or
+  // failed to inject the credential.
+  bool fail_request = 2;
```

Comment:
I completely agree with renaming the field to make it more explicit, however. However, I'm not certain the default behavior should be failing the requests, as this filter doesn't authenticate them. While there may be some scenarios where we want to abort the requests after failing to inject credentials, it should'nt be the default behavior.

Reply:
I believe the word `invalid` in the `invalid_credential_fails_request` is not accurate because there're two cases that can lead to a failure to inject credentials, and neither of them should be labeled as `invalid` credential:
- can't find the credential that need to be injected
- something goes wrong when injecting the credential, for example, failed to retrive credential from the oauth2 server

Reply:
My question is whether typically the upstream expects that the injected credential will appear in the proxied request, and what should it do if the request doesn't contain it.
I'm assuming that the default behavior should be that the injected credential is mandatory (otherwise, why was this filter added in the first place). Also if the upstream will reject a request that doesn't contain the injected credential, then the proxy should do the same (by default).
I suggest using a WKT type, as this allows changing the default value, and requires the management server to not assume what the default value is.

RE naming, I agree that invalid is not an optimal name. I will also add another case to the 2 you suggested: what if the credentials are there, but are invalid (either syntactically or semantically).
Another suggestion, converting negative to positive, `allow_empty_credential` (not sure if credentials can be empty in the first place, or if this captures all the cases).

Reply:
How about these names?
* fail_through
* allow_request_without_credential
* continue_on_credential_failure
* ignore_credential_error